### PR TITLE
Add test which replicates Dean's boxfill update issue.

### PR DESCRIPTION
**Pull request by dlonie**
This test triggers a reproducable failure when changing levels on a boxfill plot.

*DO NOT MERGE* This just replicates the bug. It does not provide a fix nor a valid baseline.

Refs #1198.

### CODE REVIEW

**Comment by dlonie** [comment]
@aashish24 Here's the script you were looking for.

@doutriaux1 @chaosphere2112 This reproduces #1198.

**Comment by aashish24** [comment]
Nice! this is exactly what we need to address our issues!! thank you for doing it. @williams13 please check it out. Thanks @dlonie

**Comment by williams13** [comment]
What exactly am I looking for?

**Comment by aashish24** [comment]
@williams13 please have a look at the test code. These are the kinds of test we need to test out state changes working.

**Comment by doutriaux1** [comment]
@aashish24 @dlonie @chaosphere2112 @williams13 this branch is now ready to merge
it fixes the issue adds a bunch of tests for it and also fix/tweaks the way we were setting extensions on graphics methods. Before they were possibilities for the use rto pass values tat would not trigger the right calls.

**Comment by doutriaux1** [comment]
goes with:
https://github.com/UV-CDAT/uvcdat-testdata/pull/33

File: Packages/vcs/Lib/VTKPlots.py [hunk 1]
```diff
@@ -817,12 +817,14 @@ def plot2D(self,data1,data2,tmpl,gm,vtk_backend_grid=None,vtk_backend_geo=None):
           cot.SetInputData(vtk_backend_grid)
 
       levs = gm.levels
-      if (isinstance(gm,isoline.Gi) and numpy.allclose( levs[0],[0.,1.e20])) or numpy.allclose(levs,1.e20):
-        levs = vcs.mkscale(mn,mx)
-        if len(levs)==1: # constant value ?
-          levs = [levs[0],levs[0]+.00001]
-        Ncolors = len(levs)
-        if isinstance(gm,(isofill.Gfi,meshfill.Gfm)):
+      ## Apparently in some cases
```
Comment (dlonie) [inline Packages/vcs/Lib/VTKPlots.py]: ?

**Comment by dlonie** [comment]
New tests pass, LGTM.

File: Packages/vcs/Lib/VTKPlots.py (hunk 1 above)
Comment (aashish24) [inline Packages/vcs/Lib/VTKPlots.py]: Can we remove this line or explain it better?

File: Packages/vcs/Lib/utils.py [hunk 2]
```diff
@@ -571,15 +572,15 @@ def scriptrun(script):
...
-            except Exception,err:
-              print "failed",typ,nm,err
-    except Exception,err: #ok could not read json file maybe it is an old initial.attributes
-      if os.path.split(script)[-1] == "initial.attributes":
-        _scriptrun(script)
-      else:
-        warnings.warn("unable to source file: %si %s" % (script,err))
+            #except Exception,err:
+            #  print "failed",typ,nm,err
+    #except Exception,err: #ok could not read json file maybe it is an old initial.attributes
```
Comment (aashish24) [inline Packages/vcs/Lib/utils.py]: If code is not useful anymore, I would suggest we remove it. Un commenting a lot of code is not desired.

File: Packages/vcs/Lib/VTKPlots.py (hunk 1 above)
Comment (doutriaux1) [inline Packages/vcs/Lib/VTKPlots.py]: Looks like old code not completely removed, will remove in a separate PR

File: Packages/vcs/Lib/utils.py (hunk 2 above)
Comment (doutriaux1) [inline Packages/vcs/Lib/utils.py]: it is actually useful, was commented out while debugging to better catch error message/traceback. Will submit a new PR right now to put it back in

Comment (doutriaux1) [inline Packages/vcs/Lib/utils.py]: @aashish24 it's in https://github.com/UV-CDAT/uvcdat/tree/boxfill_bug_branch_clean_up
But there is still an issue with boxfill_bug in GUI. I might need to tweak this branch some more, so I'm not issuing the PR yet.

Comment (aashish24) [inline Packages/vcs/Lib/utils.py]: thanks. just wanted to make sure. Like I said if code is not useful remove it. Commenting a big chunk of code makes code unreadable and non-production like.

**Pull request merged**